ch7/exercise7-1/bytecounter.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"bufio"
+	"bytes"
+)
+
+type ByteCounter int
+type LineCounter int
+type WordCounter int 
+
+
+func (c *ByteCounter) Write(p []byte) (int, error) {
+	*c += ByteCounter(len(p)) // convert int to ByteCounter
+	return len(p), nil
+}
+
+
+
+func (c *WordCounter) Write(p []byte) (int, error) {
+	var count int
+	buf := bytes.NewBuffer(p)
+	scanner := bufio.NewScanner(buf)
+	scanner.Split(bufio.ScanWords)
+
+	for scanner.Scan() {
+		count++
+	}
+
+	if err := scanner.Err(); err != nil {
+        return 0, err
+	}
```

Comment:
There is still something up with your vscode `gofmt`.  Notice the wildly different indentation levels here. Instead of repeatedly mentioning it, I put up https://github.com/quinqu/tgpl/pull/19 to have robots automatically enforce standards.

Unfortunately, I don't know what you need to do to configure vscode correctly since I'm not a user. However I'm pretty positive it can be set to automatically `gofmt` code.  Perhaps you can ask other vscode users at Gravitational for assistance?